src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -177,15 +177,16 @@ public static Unsafe getUnsafe() {
     @IntrinsicCandidate
     public native void putInt(Object o, long offset, int x);
 
-    private static final int JVM_ACC_FIELD_INLINED = 0x00008000; // HotSpot-specific bit
 
     /**
      * Returns true if the given field is flattened.
      */
     public boolean isFlattened(Field f) {
-        return (f.getModifiers() & JVM_ACC_FIELD_INLINED) == JVM_ACC_FIELD_INLINED;
+        return isFlattenedField0(f);
```

Comment:
It should do a null check before calling the native entry:

```
    if (f == null) {
         throw new NullPointerException();
    }
```

Reply:
Mandy,

Thank you for the review.
I've integrated the null check before pushing the changeset.

Fred